api/turing/api/appcontext.go

```diff
@@ -81,19 +85,50 @@ func NewAppContext(
 		return nil, errors.Wrapf(err, "Failed initializing cluster controllers")
 	}
 
+	// Initialise Batch components
+	// Since there is only the default environment, we will not create multiple batch runners.
+	ensemblingImageBuilder, err := imagebuilder.NewEnsemberJobImageBuilder(
+		clusterControllers[cfg.EnsemblingJobConfig.DefaultEnvironment],
+		cfg.EnsemblingJobConfig.ImageConfig,
+		cfg.EnsemblingJobConfig.KanikoConfig,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed initializing ensembling image builder")
+	}
+
+	ensemblingJobService := service.NewEnsemblingJobService(db, cfg.EnsemblingJobConfig.DefaultEnvironment)
+	batchEnsemblingController := batchcontroller.NewBatchEnsemblingController(
+		clusterControllers[cfg.EnsemblingJobConfig.DefaultEnvironment],
+		mlpSvc,
+		cfg.SparkInfraConfig,
+	)
+
+	batchEnsemblingJobRunner := batchrunner.NewBatchEnsemblingJobRunner(
+		batchEnsemblingController,
+		ensemblingJobService,
+		mlpSvc,
+		ensemblingImageBuilder,
+		cfg.EnsemblingJobConfig.InjectGojekConfig,
```

Comment:
https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/batch/runner/ensembling.go#L281

I'm not sure of a better way to do this because this is extremely specific to our use case, but I don't think we should be injecting these labels all the time if I were an external user.

It seems to be hard coded here as well (but this forces the user to use our `gojek.com/<something>` labels): https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/cluster/servicebuilder/service_builder.go#L19-L28

Do you have any suggestions for this? Ideally we don't want these labels for external parties as well.

Reply:
I see. First, `cfg.EnsemblingJobConfig.InjectGojekConfig` is a misleading value for the parameter named `injectGojekLabels`.  So at least the config key should be `cfg.EnsemblingJobConfig.InjectGojekLabels`.
Second, I see two possible improvements:

1. You can parametrize labels's prefix (`gojek.com/`) via a config key (with a default value still being `gojek.com/`)
2. You can configure all labels altogether with a config key `cfg.EnsemblingJobConfig.Labels`, but then you need to support some sort of a meta values, that are dynamically extracted from the context. I mean something like this:
```yaml
EnsemblingJobConfig:
    Labels:
    - gojek.com/team: {{ .Metadata.Project.Team }} 
    - gojek.com/stream: {{ .Metadata.Project.Stream }} 
    - ...
```
So users can configure what labels do they want to use (if any) and we can programatically replace `{{ .Metadata.Project.Team }}` with an actual value extracted from the MLP project. This could be an overkill though, so I'd just go with an option #1.

Reply:
Ok, if thats the case i will refactor all the code that uses this `gojek.com/` prefix. As what @krithika369 mentioned here: https://github.com/gojek/turing/pull/61#discussion_r654933416

Reply:
I have refactored everything that uses the `gojek.com/...` labels.